src/app/[locale]/projects/components/DeleteProjectDialog.tsx

```diff
@@ -0,0 +1,216 @@
+import { Alert, Button, Form, Modal } from 'react-bootstrap'
+import { AiOutlineQuestionCircle } from 'react-icons/ai'
+
+const DEFAULT_PROJECT_DATA: Project = {
+    name: '',
+    _embedded: {
+        'sw360:releases': [],
+        'sw360:projects': [],
+        'sw360:attachments': [],
+    },
+}
+interface Data {
+    attachment?: number
+    project?: number
+    release?: number
+}
+
+interface Props {
+    projectId?: string
+    show?: boolean
+    setShow?: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+const DeleteProjectDialog = ({ projectId, show, setShow }: Props) => {
+    const { data: session } = useSession()
+    const t = useTranslations('default')
+    const router = useRouter()
+    const [project, setProject] = useState<Project>(DEFAULT_PROJECT_DATA)
+    const [internalData, setInternalData] = useState<Data>({ attachment: 0, project: 0, release: 0 })
+    const [variant, setVariant] = useState('success')
+    const [message, setMessage] = useState('')
+    const [showMessage, setShowMessage] = useState(false)
+    const [reloadPage, setReloadPage] = useState(false)
+    const [hasSubProject, setHasSubProject] = useState(false)
+    const [comment, setComment] = useState('')
+
+    const displayMessage = (variant: string, message: string) => {
+        setVariant(variant)
+        setMessage(message)
+        setShowMessage(true)
+    }
+
+    const handleError = useCallback(() => {
+        displayMessage('danger', t('Error when processing'))
+        setReloadPage(true)
+    }, [t])
+
+    const deleteProject = async () => {
+        const response = await ApiUtils.DELETE(`projects/${projectId}`, session.user.access_token)
+        try {
+            if (response.status == HttpStatus.OK) {
+                displayMessage('success', t('Delete project successful!'))
+                router.push('/projects')
+                setReloadPage(true)
+            } else if (response.status == HttpStatus.ACCEPTED) {
+                displayMessage('info', t('Moderation request is created!'))
+            } else if (response.status == HttpStatus.CONFLICT) {
+                displayMessage('danger', t('The project cannot be deleted, since it is used by another project!'))
+            } else if (response.status == HttpStatus.UNAUTHORIZED) {
+                await signOut()
+            } else {
+                displayMessage('danger', t('Error when processing'))
+            }
+        } catch (err) {
+            handleError()
+        }
+    }
+
+    const fetchData = useCallback(
+        async (signal: AbortSignal) => {
+            if (session) {
+                const projectsResponse = await ApiUtils.GET(`projects/${projectId}`, session.user.access_token, signal)
+                if (projectsResponse.status == HttpStatus.OK) {
+                    const project = (await projectsResponse.json()) as Project
+                    setProject(project)
+                    handleInternalDataCount()
```

Comment:
If you use this function after setProject function, this function may not receive the newest state.
Because set state is asynchronous function, so maybe **handleInternalDataCount** will run before state was updated and you will handle older state.
You can set project as a parameter to solve this problem
`
handleInternalDataCount(project)
`

Reply:
@hoangnt2 , I think if we are not showing any details of the current project at delete dialog (as mentioned above as new GUI), we can remove the `handleInternalDataCount()` method.